pkg/estimator/client/cache.go

```diff
@@ -98,8 +92,14 @@ func (c *SchedulerEstimatorCache) GetClient(name string) (estimatorservice.Estim
 
 // EstablishConnection establishes a new gRPC connection with the specified cluster scheduler estimator.
 func EstablishConnection(kubeClient kubernetes.Interface, name string, estimatorCache *SchedulerEstimatorCache, estimatorServicePrefix string, grpcConfig *grpcconnection.ClientConfig) error {
+	// check if the connection already exists, if so, close it and establish a new connection
 	if estimatorCache.IsEstimatorExist(name) {
-		return nil
+		// close it and establish a new connection
+		err := estimatorCache.estimator[name].connection.Close()
+		if err != nil {
+			klog.Errorf("Failed to close connection of cluster(%s): %v.", name, err)
+			return err
+		}
```

Comment:
There may be concurrency issues here. If more than one worker have established connections at the same time, a connection may be closed multiple times.

Reply:
fixed